packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ProjectController.java

```diff
@@ -552,10 +557,42 @@ public ResponseEntity<Project> updateProject(
 			})
 	@GetMapping("/export/{id}")
 	@Secured(Roles.USER)
-	public ResponseEntity<ProjectExport> exportProject(@PathVariable("id") final UUID id) {
+	public ResponseEntity<byte[]> exportProject(@PathVariable("id") final UUID id) {
 		projectService.checkPermissionCanRead(currentUserService.get().getId(), id);
 		try {
-			return ResponseEntity.ok(cloneService.exportProject(id));
+			final ProjectExport export = cloneService.exportProject(id);
+
+			// Serialize the project export to a byte array
+			final byte[] exportBytes = objectMapper.writeValueAsBytes(export);
+
+			try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+					ZipOutputStream zipOutputStream = new ZipOutputStream(byteArrayOutputStream)) {
+
+				// Example entry. Replace with actual file content.
+				final ZipEntry zipEntry = new ZipEntry("project.json");
```

Comment:
Is it reasonable to name the zipped json file simply `.json`, should it be `.json.zip` or some such instead?